### ISSUE TITLE
Support for AWS profiles (IAM credentials and SSO)

### ISSUE THREAD

**Issue by n0ct1s-k8sh** [spacechunks/packer-plugin-s3#2]
## Background

It's known that IAM access and secret keys are sensible and discouraged to use, as the SSO profiles are implemented constantly.

## Goal

A `profile` parameter which contains the profile name configured in the AWS CLI should be implemented to authenticate and perform the operations within the AWS SDK (which I see it is not implemented).

## References
- <https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/#single-sign-on-credentials>
- <https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html#cli-configure-files-using-profiles>

## Example

```hcl
packer {
  required_plugins {
    s3 = {
      version = "1.0.0"
      source = "github.com/spacechunks/s3"
    }
  }
}

provisioner "s3" {
  profile = "<aws-cli-profile-name>"
  endpoint = "s3.amazonaws.com"
  objects {
    source = "mybucket/myfolder/somefile"
    destination = "/etc/myobject2"
  }
  objects {
    source = "mybucket2/myfolder/somefile"
    destination = "/etc/myobject2"
  }
}
```

**Comment by freggy** [comment]
hey, thanks for opening an issue :) it makes sense switching to the official AWS S3 SDK. i will implement this, but currently I am busy with other things, so I can't give you an ETA when this will be ready.

i will reach out to you, once it's implemented.

**Comment by freggy** [comment]
Hey @n0ct1s-k8sh, the AWS v2 SDK is now implemented as per https://github.com/spacechunks/packer-plugin-s3/pull/3, so feel free to test it out.

**Comment by n0ct1s-k8sh** [comment]
Good job mate! Thanks.

I'm trying to build a personal image with latest Packer (1.11.2) and it fails in the init process:
```
Failed getting the "github.com/spacechunks/s3" plugin:
2 errors occurred:
        * Continuing to next available version: binary reported version ("1.0.0") is different from the expected "2.0.0", skipping
        * could not install any compatible version of plugin "github.com/spacechunks/s3"
```

I saw the [`v2.0.0`](https://github.com/spacechunks/packer-plugin-s3/releases/tag/v2.0.0) release and I'd set up it like this:
```
packer {
  required_plugins {
    s3 = {
      version = ">~2"
      source = "github.com/spacechunks/s3"
    }
  }
}
```

**Comment by freggy** [comment]
Sorry, I forgot to bump the binary version to `2.0.0`. It is fixed now. 

Run 
```
packer plugins remove github.com/spacechunks/s3 2.0.0
```
and do packer init again and it should work.
